src/main/java/io/druid/data/input/impl/SpatialDimensionSchema.java

```diff
@@ -18,12 +18,14 @@
 package io.druid.data.input.impl;
 
 import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonIgnore;
 import com.fasterxml.jackson.annotation.JsonProperty;
 
 import java.util.List;
 
 /**
  */
+@Deprecated
 public class SpatialDimensionSchema
 {
   private final String dimName;
```

Comment:
@himanshug 

I tried that just now, I encountered a problem with that approach:
- In DimensionSchema, I set the default schema type to StringDimensionSchema, to support the old convention of specifying a dimension name

- The old SpatialDimensionSchema does not have a "type" field

- If I try to deserialize a JSON string that contains an old SpatialDimensionSchema, it tries to create a StringDimensionSchema (since type is missing), which is incorrect and also fails with a NPE since the original SpatialDimensionSchema does not have a "name" field either